### ISSUE TITLE
Build failure with jfoenix 9.0.6 on android using api level 27 / java 10

### ISSUE THREAD

**Issue by aminabs** [jfoenixadmin/JFoenix#826]
Hi there,

I am trying to build my project for android using jfoenix 9.0.6.

System spec:
**_JDK 10: java version 10.0.2
jfxmobile-plugin: version 2.0.29_**

Here is the stack trace applying the following command:
**gradle clean build androidInstall**

java.lang.RuntimeException: Failed to analyze class: 'com\jfoenix\adapters\ReflectionHelper.class'.
Classes compiled with JDK 9 or later are currently not supported on Android. Please make sure that your project does not contain JDK 9+ dependencies.
	at org.javafxports.retrobuffer.Retrobuffer$1.visitClass(Retrobuffer.java:62)
	at org.javafxports.retrobuffer.ClasspathVisitor.visitFile(ClasspathVisitor.java:59)
	at org.javafxports.retrobuffer.ClasspathVisitor.visitFile(ClasspathVisitor.java:41)
	at java.base/java.nio.file.Files.walkFileTree(Files.java:2716)
	at java.base/java.nio.file.Files.walkFileTree(Files.java:2788)
	at org.javafxports.retrobuffer.Retrobuffer.run(Retrobuffer.java:56)
	at org.javafxports.retrobuffer.Main.main(Main.java:45)
Caused by: java.lang.IllegalArgumentException
	at org.objectweb.asm.ClassReader.<init>(Unknown Source)
	at org.objectweb.asm.ClassReader.<init>(Unknown Source)
	at org.javafxports.retrobuffer.ClassAnalyzer.analyze(ClassAnalyzer.java:48)
	at org.javafxports.retrobuffer.Retrobuffer$1.visitClass(Retrobuffer.java:60)
	... 6 more

could you please help me on this?

**Comment by skmedix** [comment]
`Classes compiled with JDK 9 or later are currently not supported on Android. Please make sure that your project does not contain JDK 9+ dependencies.`

Just read the error. Use JDK 8 and JFoenix 8

**Comment by aminabs** [comment]
Hey,
For some reason I have to migrate to jdk 10 on android using gluon vm.

I am not sure if javafxport version used by jfxmobile plugin 2.x.x is still 8 or 9?

And how woyld it be reasonable if they port jdk 10 along with javafx 8 on android?!

I know this question goes back to the gluon guys, but would appritiate if anyone knows any information about it?

**Comment by jfoenixadmin** [comment]
Hello, @aminabs as far as i know jfxmobile plugin is still Java 8, and the ported java version is dalvik 8.60 along with retro-lambda library.

**Comment by aminabs** [comment]
Thank you @jfoenixadmin,

According to the documentation,  jfxmobile-plugin v2.x.x supports java 9. However, latest android does not support it yet. Android v >= 7 (api level >= 24) supports java 8. 
So there should be no need to applying retrolambda plugin.
I've tested the v2.0.29 of the plugin with my android 8 phone; It is not clear to me why the plugin still applies retrolambda on lambda functions although I've set the minSdkVersion to 24 (android 7).
